sources/common/src/XmlSignalFactory.cpp

```diff
@@ -133,37 +133,45 @@ PARSER::XmlSignalFactory::XmlSignalFactory(const std::string &rXmlFileName) :
 
     pugi::xml_node signals = doc.child("signals");
 
-    // Traverse the XML.
-    for (const pugi::xml_node &signal : signals.children())
+    try
     {
-        if (!std::strcmp("vector", signal.name()))
+        // Traverse the XML.
+        for (const pugi::xml_node &signal : signals.children())
         {
-            std::unique_ptr<XmlDescription> description = std::make_unique<XmlDescription>(signal);
```

Comment:
This giant `if` is a perfect candidate to be replaced by a factory.

Reply:
In fact it is a kind of factory method. I simplified it by just having one place in which the signal description is read. It slightly shortened the method. I'd leave it as it is.